src/Microsoft.ML.Data/Data/SchemaDefinition.cs

```diff
@@ -349,98 +340,132 @@ public static SchemaDefinition Create(Type userType, Direction direction = Direc
                 ((direction & Direction.Write) == Direction.Write && (x.CanWrite && x.GetSetMethod() != null))) &&
                 x.GetIndexParameters().Length == 0);
 
-            var memberInfos = (fieldInfos as IEnumerable<MemberInfo>).Concat(propertyInfos).ToArray();
+            return (fieldInfos as IEnumerable<MemberInfo>).Concat(propertyInfos).ToArray();
+        }
 
-            foreach (var memberInfo in memberInfos)
+        public static bool CheckMemberInfo(MemberInfo memberInfo)
+        {
+            switch (memberInfo)
             {
-                // Clause to handle the field that may be used to expose the cursor channel.
-                // This field does not need a column.
-                // REVIEW: maybe validate the channel attribute now, instead
-                // of later at cursor creation.
-                switch (memberInfo)
-                {
-                    case FieldInfo fieldInfo:
-                        if (fieldInfo.FieldType == typeof(IChannel))
-                            continue;
+                case FieldInfo fieldInfo:
+                    if (fieldInfo.FieldType == typeof(IChannel))
+                        return false;
```

Comment:
Nit: originally there was this comment right before the switch, perhaps it's worth it to keep that comment.

```
                // Clause to handle the field that may be used to expose the cursor channel.
                // This field does not need a column.
                // REVIEW: maybe validate the channel attribute now, instead
                // of later at cursor creation.
```